drivers/i2c/i2c_mcux_flexcomm.c

```diff
@@ -19,6 +19,12 @@ LOG_MODULE_REGISTER(mcux_flexcomm);
 
 #include "i2c-priv.h"
 
+#if CONFIG_I2C_NXP_TRANSFER_TIMEOUT
+#define I2C_TRANSFER_TIMEOUT_MSEC K_MSEC(CONFIG_I2C_NXP_TRANSFER_TIMEOUT)
+#else
+#define I2C_TRANSFER_TIMEOUT_MSEC K_FOREVER
+#endif
```

Comment:
Nit, you can remove #if with COND_CODE_0 macro

https://docs.zephyrproject.org/apidoc/latest/group__sys-util.html#ga5483ea38af79bc6c4509936288705377

Reply:
Before I commit, just to confirm, is this the format?
`#define I2C_TRANSFER_TIMEOUT_MSEC                                                                  \
	COND_CODE_0(CONFIG_I2C_NXP_TRANSFER_TIMEOUT, (K_FOREVER),                                  \
		    (K_MSEC(CONFIG_I2C_NXP_TRANSFER_TIMEOUT)))
`

Reply:
that's fine

Reply:
@decsny Done.